stan/math/prim/fun/matrix_exp_2x2.hpp

```diff
@@ -47,7 +47,12 @@ matrix_exp_2x2(const EigMat& A) {
   B(1, 0) = c * Two_exp_sinh;
   B(1, 1) = exp_half_a_plus_d * (delta_cosh - ad_sinh_half_delta);
 
-  return B / delta;
+  // use pade approximation if cosh & sinh ops overflow to NaN
+  if ((B.array() != B.array()).any()) {
```

Comment:
Optional, but I think Eigen 3.3.9 also supports `hasNaN()`. Might be easier to read and does not require additional comments to understand what is going on.

Reply:
TIL. Thanks!